pkg/app/piped/executor/analysis/metrics_analyzer.go

```diff
@@ -114,6 +148,152 @@ func (a *metricsAnalyzer) analyzeWithThreshold(ctx context.Context) (expected bo
 		return false, nil
 	}
 
-	a.logPersister.Successf("[%s] The query result is expected one. Performed query: %q", a.id, a.cfg.Query)
 	return true, nil
 }
+
+// analyzeWithPrevious returns false if primary deviates in the specified direction compared to the previous deployment.
+// Return an error if the evaluation could not be executed normally.
+// elapsedTime is used to compare metrics at the same point in time after the analysis has started.
+func (a *metricsAnalyzer) analyzeWithPrevious(ctx context.Context) (expected, firstDeploy bool, err error) {
+	now := time.Now()
+	queryRange := metrics.QueryRange{
+		From: now.Add(-a.cfg.Interval.Duration()),
+		To:   now,
+	}
+	points, err := a.provider.QueryPoints(ctx, a.cfg.Query, queryRange)
+	if err != nil {
+		return false, false, fmt.Errorf("failed to run query: %w", err)
+	}
+
+	prevMetadata, err := a.analysisResultStore.GetLatestAnalysisResult(ctx)
+	if errors.Is(err, analysisresultstore.ErrNotFound) {
+		return false, true, nil
+	}
+	if err != nil {
+		return false, false, fmt.Errorf("failed to fetch the most recent successful analysis metadata: %w", err)
+	}
+	// Compare it with the previous metrics when the same amount of time as now has passed since the start of the stage.
+	elapsedTime := time.Since(a.stageStartTime)
```

Comment:
This elapsedTime should be calculated by using `now` instead of a new `time.Now` like this.

Reply:
For sure!

Reply:
good catch ;)